specifyweb/frontend/js_src/lib/components/SchemaConfig/UniquenessRules.tsx

```diff
@@ -0,0 +1,467 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import type { LocalizedString } from 'typesafe-i18n';
+
+import { commonText } from '../../localization/common';
+import { schemaText } from '../../localization/schema';
+import { ajax } from '../../utils/ajax';
+import type { RA } from '../../utils/types';
+import { defined, filterArray, localized } from '../../utils/types';
+import { removeKey, sortFunction, split } from '../../utils/utils';
+import { Button } from '../Atoms/Button';
+import { className } from '../Atoms/className';
+import { icons } from '../Atoms/Icons';
+import { Submit } from '../Atoms/Submit';
+import { LoadingContext } from '../Core/Contexts';
+import { addMissingFields } from '../DataModel/addMissingFields';
+import type { SerializedResource } from '../DataModel/helperTypes';
+import { schema } from '../DataModel/schema';
+import type { RelationshipType } from '../DataModel/specifyField';
+import { getTable, strictGetTable } from '../DataModel/tables';
+import type {
+  SpLocaleContainer,
+  SpLocaleContainerItem,
+  Tables,
+} from '../DataModel/types';
+import type {
+  UniquenessRule,
+  UniquenessRuleValidation,
+} from '../DataModel/uniquenessRules';
+import {
+  getUniqueInvalidReason,
+  useModelUniquenessRules,
+  validateUniqueness,
+} from '../DataModel/uniquenessRules';
+import { raise } from '../Errors/Crash';
+import { Dialog } from '../Molecules/Dialog';
+import { userPreferences } from '../Preferences/userPreferences';
+import type { WithFetchedStrings } from '../Toolbar/SchemaConfig';
+import { downloadDataSet } from '../WorkBench/helpers';
+import { PickList } from './Components';
+import { useContainerItems } from './Hooks';
+
+const databaseFieldName = '_database';
+const databaseResourceUri = '/_database';
+const dataBaseScope: SerializedResource<SpLocaleContainerItem> &
+  WithFetchedStrings = {
+  resource_uri: databaseResourceUri,
+  name: databaseFieldName,
+  strings: { name: { text: schemaText.database() } },
+};
+
+export function TableUniquenessRules({
```

Comment:
This type of query is generally fast, but of course there are always performance concerns for significantly large databases. 

@specify/ux-testing 
If it would be preferred, we can have validation run for every rule before the rules are saved/committed to the database. 
This might be presented in the save button being a validate button and then becoming a save button, a separate button solely for validation, or implicitly running validation when the save button is clicked. 

We would need to consider the benefit of such a feature when compared to implementation time, but it would probably be favorable when doing batch-validation to do so only on the rules which have changed since the last validation.

Reply:
> Or is this type of query fast?

The major, and most significant, factor if the fields are indexed are not